llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -17271,6 +17271,10 @@ SDValue DAGCombiner::visitFSUB(SDNode *N) {
 // prefer it.
 SDValue DAGCombiner::combineFMulOrFDivWithIntPow2(SDNode *N) {
   EVT VT = N->getValueType(0);
+  if (!Type::getFloatingPointTy(*DAG.getContext(), VT.getFltSemantics())
+           ->isIEEELikeFPTy())
```

Comment:
We shouldn't have to go through the IR type here. Can't you get this directly from the fltSemantics